src/main/java/seedu/partyplanet/logic/commands/EListCommand.java

```diff
@@ -31,6 +31,18 @@
 
     public static final Comparator<Event> SORT_NAME = Comparator.comparing(x -> x.getName().fullName);
     public static final Comparator<Event> SORT_EVENTDATE = Comparator.comparing(Event::getEventDate);
+    public static final Comparator<Event> SORT_EVENTDATE_UPCOMING = (Event x, Event y) -> {
+        // Sort done events at the end
+        if (x.isDone() && !y.isDone()) {
+            return 1;
+        }
+        if (!x.isDone() && y.isDone()) {
+            return -1;
+        }
+        Long xDaysLeft = x.getEventDate().getDaysLeft();
+        Long yDaysLeft = y.getEventDate().getDaysLeft();
+        return xDaysLeft.compareTo(yDaysLeft);
```

Comment:
```suggestion
        if (xDaysLeft < 0 && yDaysLeft >= 0) {
            return 1;
        } else if (xDaysLeft < 0 && yDaysLeft >= 0) {
            return -1;
        } else {
            return xDaysLeft.compareTo(yDaysLeft);
        }
```

Reply:
Thanks for the suggestion! Implemented the alternative algorithm as separately discussed earlier, see upcoming commit.